torchgen/model.py

```diff
@@ -1194,16 +1449,121 @@ def strip_ret_annotation(r: Return) -> Return:
                 ),
                 overload_name="",  # stripped
             ),
-            arguments=self.arguments.signature(strip_default=strip_default),
-            returns=tuple(map(strip_ret_annotation, self.returns)),
+            arguments=args_sig,
+            returns=returns,
         )
 
     def view_signature(self) -> "FunctionSchema":
         return self.signature(strip_view_copy_name=True)
 
+    def self_to_out_signature(self) -> "FunctionSchema":
```

Comment:
These methods are very long but they are only called from one place, so it would make me happier if they were more private rather than being directly on FunctionSchema object. If you take my advice and move generate_function out of model.py these will naturally go too.